bin/sync_schema

```diff
@@ -0,0 +1,211 @@
+    when :date
+      append_presence_check("xml.#{xml_name} attributes.#{name}.strftime(\"%m/%d/%Y\")")
+    # when :struct
+      # binding.pry
+    else
+      append_presence_check("xml.#{xml_name} attributes.#{name}")
+    end
+  end
+
+  private
+
+  attr_reader :ref_name, :required
+
+  def custom_field_ruby(action)
+    <<-RUBY
+      if attributes.customfields.present? || attributes.to_h.except(*#{action.upcase}_KEYS).present?
+        xml.customfields do
+          if attributes.customfields.present?
+            attributes.customfields.presence&.each do |custom_field|
+              xml.customfield do
+                xml.customfieldname customfield[:customfieldname]
+                xml.customfieldvalue customfield[:customfieldvalue]
+              end
+            end
+          end
+
+          if attributes.to_h.except(*#{action.upcase}_KEYS).present?
+            attributes.to_h.except(*#{action.upcase}_KEYS).each do |name, value|
+              xml.customfield do
+                xml.customfieldname name
+                xml.customfieldvalue value
+              end
+            end
+          end
+        end
+      end
+    RUBY
+  end
+
+  def custom_fields?
+    name == "customfields"
+  end
+
+  def append_presence_check(line)
+    if required?
+      line
+    else
+      "#{line} if attributes.#{name}.present?"
+    end
+  end
+
+  def wrap_presence_check(multiline)
+    <<-RUBY
+      if attributes.#{name}.present?
+        #{multiline}
+      end
+    RUBY
+  end
+
+  def date?
+    sequence&.count == 3 && sequence.map(&:name).sort == %w[day month year]
+  end
+
+  def struct?
+    sequence.present?
+  end
+
+  def build_sequence(element)
+    complex = filtered_children(element).detect { |child| child.name == "complexType" }
+    return unless complex.present?
+
+    return if complex.attributes.values.any? { |child| child.name == "mixed" && child.value == "true" }
+
+    sequence = filtered_children(complex).detect { |child| child.name == "sequence" }
+    return if sequence.blank?
+
+    @sequence = filtered_children(sequence).map { |child| XmlType.new(child) }
+  end
+end
+
+filtered_children(parsed.root).map do |child|
+  if XmlType.type?(child) || XmlType.ref?(child)
+    type = XmlType.new(child)
+    RegisteredTypes.xml_types[type.type_name] = type
+  end
+end.compact
+
+def process_template(object)
+  @class_name = object.camelcase
+  @create_attributes = RegisteredTypes.xml_types["create_#{object}"].sequence
+  @update_attributes = RegisteredTypes.xml_types["update_#{object}"].sequence
+
+  @create_names = @create_attributes.map { |attr| attr.name.to_sym }
+  @update_names = @update_attributes.map { |attr| attr.name.to_sym }
+
+  if @identical = @create_names == @update_names
+    @update_names = "CREATE_KEYS"
+  end
+
+  File.write(File.expand_path("../lib/intacct/models/#{object}.rb", __dir__), ERB.new(@template, trim_mode: "-").result)
+end
+
+%w[project task].each { |object| process_template(object) }
```

Comment:
Would it make sense to break these out into a constant or somewhere more visible?